core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -62,44 +68,35 @@ public String getFullPath() {
     }
 
 
-    public int  getLikes() {
-        int likes = 0;
 
+    @PostConstruct
+    private void init() {
         String rootPath = resource.getPath();
-
-        resource = resourceResolver.getResource(rootPath);
-        ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
-
-        if (properties != null) {
-            likes = resource.adaptTo(ValueMap.class).get("likes", 0);
-            properties.put("likes", likes + 1);
-        }
-        try {
-            resourceResolver.commit();
-        } catch (PersistenceException e) {
-            log.info(e.getMessage());
+        String dislikeParam = servletRequest.getParameter("dislike");
+        String likeParam = servletRequest.getParameter("like");
+        if (likeParam != null) {
+            likes = incrementLikeDislike(likeParam, rootPath);
```

Comment:
What happens if a user just opens the current page for the first time, without pressing neither like, nor dislike?
There is not any `like` or `dislike` parameter in the URL, and therefore the fields `likes` and `dislikes` are not initialized at all. They equal to zero. And the user see all the zeros in the page until he finally presses a "like" or "dislike" button. I believe this is wrong.